tests/wpunit/UserObjectCursorTest.php

```diff
@@ -0,0 +1,660 @@
+		for ( $i = 0; $i < $paged_count; $i ++ ) {
+			$this->assertEquals( $expected[$i], $actual['data']['users']['nodes'][$i]['userId'] );
+		}
+
+		$expected = array_slice( $this->created_user_ids, $paged_count, $paged_count );
+
+		codecept_debug( $expected );
+
+		$actual = graphql( [
+			'query' 		=> $this->query,
+			'variables' => [
+				'first' => $paged_count,
+				'after' => $actual['data']['users']['pageInfo']['endCursor'],
+				'where' => [
+					'search' => 'http://www.test.test',
+				],
+			],
+		] );
+
+		codecept_debug( $actual );
+
+		$this->assertArrayNotHasKey( 'errors', $actual );
+
+		for ( $i = 0; $i < $paged_count - 1; $i ++ ) {
+			$this->assertEquals( $expected[$i], $actual['data']['users']['nodes'][$i]['userId'] );
+		}
+	}
+
+	/**
+	 * Tests the backward pagination of connections
+	 * @throws Exception
+	 */
+	public function testUsersBackwardPagination() {
+
+		$paged_count = ceil( $this->count / 2 );
+		$expected = array_slice( $this->created_user_ids, $paged_count - 1, $paged_count );
+
+		codecept_debug( $expected );
+
+		$actual = graphql( [
+			'query' 		=> $this->query,
+			'variables' => [
+				'last' => $paged_count,
+				'where' => [
+					'search' => 'http://www.test.test',
+				],
+			],
+		] );
+
+		codecept_debug( $actual );
+
+		$this->assertArrayNotHasKey( 'errors', $actual );
+
+		// Compare actual results to ground truth
+		for ( $i = 0; $i < $paged_count; $i ++ ) {
+			$this->assertEquals( $expected[$i], $actual['data']['users']['nodes'][$i]['userId'] );
+		}
+
+		$paged_count = ceil( $this->count / 2 );
+		$expected = array_slice( $this->created_user_ids, 0, $paged_count - 1);
+
+		codecept_debug( $expected );
+
+		$actual = graphql( [
+			'query' 		=> $this->query,
+			'variables' => [
+				'last' => $paged_count,
+				'before' => $actual['data']['users']['pageInfo']['startCursor'],
+				'where' => [
+					'search' => 'http://www.test.test',
+				],
+			],
+		] );
+
+		codecept_debug( $actual );
+
+		$this->assertArrayNotHasKey( 'errors', $actual );
+
+		for ( $i = 0; $i < $paged_count - 1; $i ++ ) {
+			$this->assertEquals( $expected[$i], $actual['data']['users']['nodes'][$i]['userId'] );
+		}
+	}
+
+	private function formatNumber($num) {
+		return sprintf('%08d', $num);
+	}
+
+	private function numberToMysqlDate($num) {
+		return sprintf('2019-03-%02d', $num);
+	}
+
+	private function deleteByMetaKey( $key, $value ) {
+		$args = array(
+			'meta_query' => array(
+				array(
+					'key' => $key,
+					'value' => $value,
+					'compare' => '=',
+				)
+			)
+		 );
+
+		 $query = new WP_User_Query($args);
+
+		 foreach ( $query->results as $user ) {
+			wp_delete_user( $user->ID, true );
+		 }
+	}
+
+	/**
+	 * Assert given 'orderby' field by comparing the GraphQL query against a plain WP_User_Query
+	 * @throws Exception
+	 */
+	public function assertQueryInCursor( $meta_fields ) {
+
+		add_filter( 'graphql_map_input_fields_to_wp_user_query', function( $query_args ) use ( $meta_fields ) {
```

Comment:
It looks like the intent of this is to have the $meta_fields input apply to the query as `$where` args. 

Instead of filtering like this, we could defined the `$where` variable and pass the input to the query as a variable. 

I think this will make the tests a bit easier to read long term.

Reply:
Looks like this was done because the actual query from GraphQL and the expected query from WP_User_Query use different args. But for the sake of readability and more direct testing, I agree we should pass the args directly into the $where args. Working on this.